ultranest/popstepsampler.py

```diff
@@ -688,7 +688,333 @@ def __next__(
+       can be adapted with the `scale_adapt_factor` parameter based on the median distance 
+       between the current and the next point in a chains among all the chains. If the median
+       distance is above `scale/adapt_slice_scale_target`, the scale is increased by `scale_adapt_factor`,
+       and decreased otherwise. The `scale` parameter can also be jittered by a user supplied
+       function `:py:func:scale_jitter_func` to counter balance the effect of a strong adaptation.
+
+       In the case `scale!=None`, the detailed balance is not guaranteed, so this sampler should
+       be use with caution.
+
+       Multiple (`popsize`) slice sampling chains are run independently and in parallel. 
+       In that case, we read points as if they were the next selected each after the other.
+       For a points to update the slice, it needs to be still in the part of the slices
+       searched after the first point have been read. In that case, we update as normal, 
+       otherwise we discard the point.
+
+
+    """
+
+    def __init__(
+        self, popsize, nsteps, generate_direction,
+        scale_adapt_factor=1.0, adapt_slice_scale_target=2.0,
+        scale=None, scale_jitter_func=None,slice_limit=slice_limit_to_unitcube,
+        max_it=100,shrink_factor=1.0):
+        """Initialise.
+
+        Parameters
+        ----------
+        popsize: int
+            number of walkers to maintain.
+        nsteps: int
+            number of steps to take until the found point is accepted as independent.
+            To calibrate, try several runs with increasing nsteps (doubling).
+            The ln(Z) should become stable at some value.
+        generate_direction: function
+            Function that gives proposal kernel shape, one of:
+            :py:func:`ultranest.popstepsampler.generate_random_direction`
+            :py:func:`ultranest.popstepsampler.generate_region_oriented_direction`
+            :py:func:`ultranest.popstepsampler.generate_region_random_direction`
+            :py:func:`ultranest.popstepsampler.generate_differential_direction`
+            :py:func:`ultranest.popstepsampler.generate_mixture_random_direction`
+            :py:func:`ultranest.popstepsampler.generate_cube_oriented_direction` -> no adaptation in that case
+            :py:func:`ultranest.popstepsampler.generate_cube_oriented_direction_scaled` -> no adaptation in that case
+        scale_jitter_func: function
+            User supplied function to multiply the `scale` by a random factor. For example, 
+            :py:func:`lambda : scipy.stats.truncnorm.rvs(-0.5, 5., loc=0, scale=1)+1.`
+        scale: float
+            initial guess for the slice width. If None, the slice width is defined as the
+            intersection between itself and the unit cube.
+        scale_adapt_factor: float
+            if 1, no adapting is done.
+            if <1, the scale is increased if the slice final size is under 1/2 the scale
```

Comment:
1/2 is not accurate here because of the adapt_slice_scale_target?

Currently the adaptation and scale behaviour is described twice, once in the class and once in the arguments here. It would less confusing if it can be described clearly in one location only. Sometimes this is possible by ordering the arguments so that it explains first the simple behaviour. For example, move scale first: 

        scale: float
            initial guess for the slice width.
            
        scale_jitter_func: function
            User supplied function to multiply the `scale` by a random factor. For example, 
            :py:func:`lambda : scipy.stats.truncnorm.rvs(-0.5, 5., loc=0, scale=1)+1.`

        scale_adapt_factor: float
            adaptation of `scale`. If 1: no adaptation. if <1, the scale is increased/decreased by this factor if the final slice length is shorter/longer than the `adapt_slice_scale_target*scale`.

        adapt_slice_scale_target: float
             Targeted ratio of the median distance between slice mid and final point among all chains of `scale`. Default: 2.0. Higher values are more conservative, lower values are faster.